src/Maestro/tests/DependencyUpdateErrorProcessor.Tests/DepenedencyUpdateErrorProcessorTests.cs

```diff
@@ -0,0 +1,94 @@
+        protected readonly ServiceProvider Provider;
+        protected readonly IServiceScope Scope;
+        protected readonly MockReliableStateManager StateManager;
+        protected readonly Mock<GitHubClient> GithubClient;
+
+        public DependencyUpdateErrorProcessorTests()
+        {
+            var services = new ServiceCollection();
+            StateManager = new MockReliableStateManager();
+            Env = new Mock<IHostingEnvironment>(MockBehavior.Strict);
+            GithubClient = new Mock<GitHubClient>(null);
+            services.AddSingleton(Env.Object);
+            services.AddSingleton<IReliableStateManager>(StateManager);
+            services.AddLogging();
+            services.AddDbContext<TestBuildAssetRegistryContext>(
+                options => { options.UseInMemoryDatabase("BuildAssetRegistry"); });
+            services.AddSingleton<Func<string,GitHubClient>>(
+             repo =>
+             {
+                 return GithubClient.Object;
+             });
+            services.Configure<DependencyUpdateErrorProcessorOptions>(
+                (options) =>
+                {
+                    options.IsEnabled = true;
+                    options.FyiHandle = "@epananth";
+                    options.GithubUrl = "https://github.com/maestro-auth-test/maestro-test2";
+                });
+            Provider = services.BuildServiceProvider();
+            Scope = Provider.CreateScope();
+            _context = new Lazy<TestBuildAssetRegistryContext>(GetContext);
+            SetUp();
+        }
+
+        public TestBuildAssetRegistryContext Context => _context.Value;
+
+        private TestBuildAssetRegistryContext GetContext()
+        {
+            return Scope.ServiceProvider.GetRequiredService<TestBuildAssetRegistryContext>();
+        }
+
+        public void Dispose()
+        {
+            Env.VerifyAll();
+            Scope.Dispose();
+            Provider.Dispose();
+        }
+
+        public void SetUp()
+        {
+            var repositoryBranchUpdate = new RepositoryBranchUpdateHistoryEntry()
+            {
+                Repository = "https://github.com/maestro-auth-test/maestro-test2",
+                Branch = "38",
+                Method = "ProcessPendingUpdatesAsync",
+                Success = false,
+                Timestamp = new DateTime(2021, 04, 01)
+            };
+
+            var installationId = Context.GetInstallationId("TestValue");
+            string token = "TestToken";
+            Mock<GitHubAppTokenProvider> githubProvider = new Mock<GitHubAppTokenProvider>();
+            githubProvider.Setup(x => x.GetAppToken()).Returns(token);
+            Context.RepoBranchUpdateInMemory = new List<RepositoryBranchUpdateHistoryEntry>
+                {repositoryBranchUpdate};
+            DependencyUpdateErrorProcessor errorProcessor =
+                ActivatorUtilities.CreateInstance<DependencyUpdateErrorProcessor>(Scope.ServiceProvider,
```

Comment:
Any reason we don't use ServiceProvider.GetService<DependencyUpdateErrorProcessor> instead of the weird ActivatorUtilities thing?

Reply:
The type isn't in the service collection, we could put it in there though.

Reply:
Ooo, interesting.  I hadn't considered resolving something with the collection that wasn't in the collection.  I didn't even know that was possible.